tests/testAirPressure.cpp

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: GPL-2.0
+#include "testAirPressure.h"
+#include "core/divesite.h"
+#include "core/divelist.h"
+#include "core/file.h"
+#include <QString>
+#include <core/qthelper.h>
+
+void TestAirPressure::initTestCase()
+{
+	/* we need to manually tell that the resource exists, because we are using it as library. */
+	Q_INIT_RESOURCE(subsurface);
+}
+
+void TestAirPressure::get_dives()
+{
+	struct dive *dive;
+	verbose = 1;
+
+	QCOMPARE(parse_file(SUBSURFACE_TEST_DATA "/dives/TestAtmPress.xml", &dive_table, &trip_table, &dive_site_table), 0);
+	dive = get_dive(0);
+	dive->selected = true;
+	QVERIFY(dive != NULL);
+}
+
+void TestAirPressure::testReadAirPressure()
+{
+	struct dive *dive;
+	dive = get_dive(0);
+	dive->selected = true;
+	QVERIFY(dive != NULL);
+	QCOMPARE(QString::number(1012), QString::number(dive->surface_pressure.mbar));
+	dive = get_dive(1);
+	dive->selected = true;
+	QVERIFY(dive != NULL);
+	if (dive->surface_pressure.mbar)
+	QCOMPARE(QString::number(991), QString::number(dive->surface_pressure.mbar));
+}
+
+void TestAirPressure::testWriteReadBackAirPressure()
+{
+	struct dive *dive;
+	int32_t ap = 1111;
+	dive = get_dive(0);
+	dive->selected = true;
+	QVERIFY(dive != NULL);
+	dive->surface_pressure.mbar = ap;
+	QCOMPARE(save_dives("./testout.ssrf"), 0);
+	QCOMPARE(parse_file("./testout.ssrf", &dive_table, &trip_table, &dive_site_table), 0);
```

Comment:
since you didn't clear the data in between, this actually tests what happens when you merge the file into itself. Maybe not what you had in mind?